src/Application/Fees/Calculator.php

```diff
@@ -118,11 +120,9 @@ private function getCoreFee(): string
             // a fee on Gift Aid. May vary by card type & country.
 
             $currencyCode = strtoupper($this->currencyCode); // Just in case (Stripe use lowercase internally).
-            if (array_key_exists($currencyCode, self::STRIPE_FEES_FIXED)) {
-                $feeAmountFixed = self::STRIPE_FEES_FIXED[$currencyCode];
-            } else {
-                $feeAmountFixed = self::STRIPE_FEES_DEFAULT;
-            }
+            // Currency code has been compulsory for some time.
+            \assert(array_key_exists($currencyCode, self::STRIPE_FEES_FIXED));
```

Comment:
I'd like to do
```suggestion
            Assertion::keyExists(self::STRIPE_FEES_FIXED, $currencyCode);
```

But that seems to give me Psalm warnings of https://psalm.dev/docs/running_psalm/issues/ImpureMethodCall/ – even if I take `#[Pure]` out of this class entirely.

Reply:
@NoelLH I think it's because of the `@psalm-immutable` on the class, psalm expects all methods to be pure. Maybe `\Assert\Assertion::keyExists` should be marked as pure, I can't remember whether throwing is considered pure or not.

But in this case I think I'd just do:

```suggestion
            /** @psalm-suppress ImpureMethodCall */
            Assertion::keyExists(self::STRIPE_FEES_FIXED, $currencyCode);
```

We know that keyExists is pure enough for our purposes.